### PR TITLE
Add presubmit docker check

### PR DESCRIPTION
**Pull request by howardjohn**
This verifies that the build won't be broken

Also move the postsubmit over to the gen code.

Depends on https://github.com/istio/tools/pull/572

### CODE REVIEW

**Review by sdake — COMMENTED** [review]
LGTM once dependent PR merges.

**Review by geeknoid — APPROVED** [review]

**Pull request merged**
